beacon_node/beacon_chain/src/validator_custody.rs

```diff
@@ -344,90 +336,92 @@ mod tests {
             spec.custody_requirement,
             "advertised custody count should be minimum spec custody requirement"
         );
+    }
 
-        // add 1 validator
-        custody_context.register_validator::<E>(vec![(0, 32_000_000_000)], Slot::new(0), &spec);
-
-        assert_eq!(
-            custody_context.custody_group_count(&spec),
-            spec.validator_custody_requirement + spec.custody_requirement,
-            "head custody count should increase with 1 validator"
-        );
-
-        assert_eq!(
-            custody_context.advertised_custody_group_count(&spec),
-            spec.custody_requirement,
-            "advertised custody count should not change"
-        );
-
-        // add 7 more validators to reach `validator_custody_requirement`
-        custody_context.register_validator::<E>(
-            vec![
-                (1, 32_000_000_000),
-                (2, 32_000_000_000),
-                (3, 32_000_000_000),
-                (4, 32_000_000_000),
-                (5, 32_000_000_000),
-                (6, 32_000_000_000),
-                (7, 32_000_000_000),
-            ],
-            Slot::new(0),
-            &spec,
-        );
+    #[test]
+    fn register_single_validator_should_update_cgc() {
+        let custody_context = CustodyContext::new(false);
+        let spec = E::default_spec();
+        let bal_per_additional_group = spec.balance_per_additional_custody_group;
+        let min_val_custody_requirement = spec.validator_custody_requirement;
+        let validators_and_expected_cgc = vec![
+            (
+                vec![(0, 1 * bal_per_additional_group)],
+                min_val_custody_requirement,
+            ),
+            (
+                vec![(0, 8 * bal_per_additional_group)],
+                min_val_custody_requirement,
+            ),
+            (vec![(0, 10 * bal_per_additional_group)], 10),
+        ];
+
+        // Update validator every epoch and assert updates
+        for (idx, (validators_and_balance, expected_cgc)) in
+            validators_and_expected_cgc.into_iter().enumerate()
+        {
+            let epoch = Epoch::new(idx as u64);
+            custody_context.register_validators::<E>(
+                validators_and_balance,
+                epoch.start_slot(E::slots_per_epoch()),
+                &spec,
+            );
 
-        assert_eq!(
-            custody_context.custody_group_count(&spec),
-            spec.validator_custody_requirement + spec.custody_requirement,
-            "head custody count should should be same as with 1 validator"
-        );
+            assert_eq!(custody_context.custody_group_count(&spec), expected_cgc);
+        }
+    }
 
-        assert_eq!(
-            custody_context.advertised_custody_group_count(&spec),
-            spec.custody_requirement,
-            "advertised custody count should not change"
-        );
+    #[test]
+    fn register_multiple_validators_should_update_cgc() {
```

Comment:
Note that this test actually register different validator each epoch, so it confirms that we don't decrease the validator custody - i should probably add a separate test to cover what i originally intended to cover (multiple validators at the same time, not different epochs)